Expensify/App
# Fix signin redirecting and some clean up

### Details

Changes are made as discussed in these comments :
https://github.com/Expensify/App/issues/4507#issuecomment-895157217
https://github.com/Expensify/App/issues/4507#issuecomment-899696045
https://github.com/Expensify/App/issues/4507#issuecomment-902316107
https://github.com/Expensify/App/issues/4507#issuecomment-902915462

### Fixed Issues

$  https://github.com/Expensify/App/issues/4507

### QA Steps 



  1.  Your language should be set to Spanish
 2.   Go to signin page
  3.  Enter any email address or phone number
   4. Try to go back to the signin page by clicking on the link below
   6. Language will be set to Spanish (and with a delay for Android)

### Tested On


### Screenshots


#### Web


https://user-images.githubusercontent.com/83179295/130502663-2776d3dc-4b0b-4d83-a864-0f68021ac4d5.mp4


#### Mobile Web


https://user-images.githubusercontent.com/83179295/131913821-d4864d21-cf7f-43b7-9f3a-e17ab7d79b57.mp4


#### Desktop


#### iOS


#### Android

https://user-images.githubusercontent.com/83179295/131913567-1ee5cfb7-ab0c-4db8-ae9d-c1718127ce98.mp4
